CAAppKitBridge/Source/NSView+CAMethods.m

```diff
@@ -120,56 +120,21 @@ - (CALayer *) makeBackingLayer
   return [CALayer layer];
 }
 
-
-/* methods from libs-gui/Headers/AppKit/NSOpenGLView.h */
-
-
-/**
- detach from the current context.  You should call it before releasing this 
- object.
-*/
-- (void) clearGLContext
-{
-  if (self->_coreAnimationData == nil) {
-      return;
-  }
-  GSCAData *currGSCAData = self->_coreAnimationData;
-  NSOpenGLContext *currGlContext = currGSCAData->_GLContext;
-if (currGlContext)
-  {
-    [currGlContext clearDrawable];
-    DESTROY(currGlContext);
-    currGSCAData->_prepared = NO;
-  }
-}
-
-- (void) setOpenGLContext: (NSOpenGLContext*)context
-{
-  GSCAData *currGSCAData = self->_coreAnimationData;
-  NSOpenGLContext *currGlContext = currGSCAData->_GLContext;
-  if ( context != currGlContext )
-      {
-      [self clearGLContext];
-      ASSIGN(currGlContext, context);
-      }
-}
-
-- (NSOpenGLContext*) openGLContext
+- (NSOpenGLContext*) _gsCreateOpenGLContext
 {
   GSCAData *currGSCAData = self->_coreAnimationData;
-  NSOpenGLContext *currGlContext = currGSCAData->_GLContext;
-  if (currGlContext == nil)
-  {
-    NSOpenGLContext *context = [[NSOpenGLContext alloc] 
-                                   initWithFormat: currGSCAData->_pixelFormat
-                                   shareContext: nil];
-
-    [self setOpenGLContext: context];
-    [context setView: self];
+  NSOpenGLContext *currGLContext = currGSCAData->_GLContext;
+  if (currGLContext == nil)
+    {
+      NSOpenGLContext *context = [[NSOpenGLContext alloc] 
+                                     initWithFormat: currGSCAData->_pixelFormat
+                                     shareContext: nil];
 
-    RELEASE(context);
-  }
-  return currGlContext;
+      ASSIGN(currGLContext, context);
+      [context setView: self];
+      RELEASE(context);
```

Comment:
So this is wrong. 

- You are allocating a context so refcount of `context` = 1.
- You are assigning `currGLContext = context` (which is what `ASSIGN` is).
- `currGLContext` has refcount of `1` because it is exactly the same object as `context`
- You are telling `context`that its view is `self`
- You are then releasing `context`, bringing its refcount to 0, and bringing `currGLContext`'s refcount to 0, thus deallocating the backing object (and making both pointers invalid)
- Why did I not mention `currGSCAData->_GLContext`? Because throughout this entire exercise, it is still ... `nil`. But were you to assign value of `context`/`currGLContext` (which is identical), I would raise even more fuss, because after that `RELEASE`, `_GLContext_ would be an invalid pointer.

Why not use `@property` for `_GLContext`? Call the property `GLContext`, thus generating methods `GLContext` and `setGLContext:`. Make the `@property` a `retain` property, so setter does the right thing with increasing the retaincount by one. Then use getter and setter instead of directly accessing the variable (which you can, because I recognize this class may want to directly mess with `GSCAData` class members; but often that's a bad practice).

Reply:
Yes, this is a very stupid bug. Sorry.

>Why not use @property for _GLContext? Call the property GLContext, thus generating methods GLContext and setGLContext
You told me not to do that because of the possible naming conflicts.